### ISSUE TITLE
internal import issues with `libv4l` feature

### ISSUE THREAD

**Issue by onkoe** [raymanfx/libv4l-rs#108]
I was having trouble with the default feature setting (`v4l2`). When I tried to use the `libv4l` feature instead, it seems that some necessary internals were configured out. (I suggest setting up CI with `cargo-all-features` - helps a lot with these kinds of problems!)

In any case, I'm using Fedora 40 with the newest packages. The Fedora package `libv4l-devel` is installed. Please let me know if you need any more info about the system configuration.

Here's the issue:  

Cargo.toml
```toml
[package]
name = "test_v4l_backend"
version = "0.1.0"
edition = "2021"

[dependencies]
anyhow = "1.0.86"
image = "0.25.2"
tracing = "0.1.40"
tracing-subscriber = "0.3.18"
v4l = { version = "0.14.0", features = ["libv4l"] }

```

`cargo run` output:
```console
barrett@farts ~/D/p/s/test_v4l_backend (main) [101]> cargo clean
     Removed 1961 files, 925.5MiB total
barrett@farts ~/D/p/s/test_v4l_backend (main)> cargo run
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.12
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.3.0
   Compiling memchr v2.7.4
   Compiling minimal-lexical v0.2.1
   Compiling glob v0.3.1
   Compiling rustix v0.38.35
   Compiling prettyplease v0.2.22
   Compiling libc v0.2.158
   Compiling regex-syntax v0.8.4
   Compiling linux-raw-sys v0.4.14
   Compiling lazy_static v1.5.0
   Compiling bitflags v1.3.2
   Compiling bitflags v2.6.0
   Compiling bindgen v0.65.1
   Compiling libloading v0.8.5
   Compiling either v1.13.0
   Compiling home v0.5.9
   Compiling crossbeam-utils v0.8.20
   Compiling once_cell v1.19.0
   Compiling log v0.4.22
   Compiling shlex v1.3.0
   Compiling lazycell v1.3.0
   Compiling peeking_take_while v0.1.2
   Compiling rustc-hash v1.1.0
   Compiling simd-adler32 v0.3.7
   Compiling anyhow v1.0.86
   Compiling thiserror v1.0.63
   Compiling adler2 v2.0.0
   Compiling num-traits v0.2.19
   Compiling lock_api v0.4.12
   Compiling clang-sys v1.8.1
   Compiling paste v1.0.15
   Compiling rayon-core v1.12.1
   Compiling built v0.7.4
   Compiling aligned-vec v0.5.0
   Compiling nom v7.1.3
   Compiling scopeguard v1.2.0
   Compiling arrayvec v0.7.6
   Compiling miniz_oxide v0.8.0
   Compiling quote v1.0.37
   Compiling crossbeam-epoch v0.9.18
   Compiling crc32fast v1.4.2
   Compiling rav1e v0.7.1
   Compiling syn v2.0.76
   Compiling adler v1.0.2
   Compiling crossbeam-deque v0.8.5
   Compiling bytemuck v1.17.1
   Compiling simd_helpers v0.1.0
   Compiling flate2 v1.0.33
   Compiling miniz_oxide v0.7.4
   Compiling spin v0.9.8
   Compiling itertools v0.12.1
   Compiling tracing-core v0.1.32
   Compiling maybe-rayon v0.1.1
   Compiling num-integer v0.1.46
   Compiling v_frame v0.3.8
   Compiling imgref v1.10.1
   Compiling new_debug_unreachable v1.0.6
   Compiling weezl v0.1.8
   Compiling bitstream-io v2.5.3
   Compiling noop_proc_macro v0.3.0
   Compiling num-bigint v0.4.6
   Compiling smallvec v1.13.2
   Compiling quick-error v2.0.1
   Compiling regex-automata v0.4.7
   Compiling loop9 v0.1.5
   Compiling flume v0.11.0
   Compiling rgb v0.8.48
   Compiling avif-serialize v0.8.1
   Compiling zune-inflate v0.2.54
   Compiling which v4.4.2
   Compiling fdeflate v0.3.4
   Compiling half v2.4.1
   Compiling cexpr v0.6.0
   Compiling zune-core v0.4.12
   Compiling overload v0.1.1
   Compiling jpeg-decoder v0.3.1
   Compiling lebe v0.5.2
   Compiling byteorder-lite v0.1.0
   Compiling color_quant v1.1.0
   Compiling bit_field v0.10.2
   Compiling zune-jpeg v0.4.13
   Compiling png v0.17.13
   Compiling nu-ansi-term v0.46.0
   Compiling image-webp v0.1.3
   Compiling gif v0.13.1
   Compiling exr v1.72.0
   Compiling tracing-log v0.2.0
   Compiling rayon v1.10.0
   Compiling qoi v0.4.1
   Compiling num-rational v0.4.2
   Compiling thread_local v1.1.8
   Compiling tiff v0.9.1
   Compiling sharded-slab v0.1.7
   Compiling pin-project-lite v0.2.14
   Compiling av1-grain v0.2.3
   Compiling regex v1.10.6
   Compiling tracing-subscriber v0.3.18
   Compiling thiserror-impl v1.0.63
   Compiling profiling-procmacros v1.0.15
   Compiling arg_enum_proc_macro v0.3.4
   Compiling num-derive v0.4.2
   Compiling tracing-attributes v0.1.27
   Compiling profiling v1.0.15
   Compiling tracing v0.1.40
   Compiling v4l-sys v0.3.0
   Compiling v4l2-sys-mit v0.3.0
   Compiling ravif v0.11.10
   Compiling image v0.25.2
   Compiling v4l v0.14.0
error[E0428]: the name `detail` is defined multiple times
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/v4l2/api.rs:59:1
   |
8  | mod detail {
   | ---------- previous definition of the module `detail` here
...
59 | mod detail {
   | ^^^^^^^^^^ `detail` redefined here
   |
   = note: `detail` must be defined only once in the type namespace of this module

error[E0252]: the name `v4l_sys` is defined multiple times
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/lib.rs:74:9
   |
71 | pub use v4l_sys;
   |         ------- previous import of the module `v4l_sys` here
...
74 | pub use v4l2_sys as v4l_sys;
   |         ^^^^^^^^^^^^^^^^^^^ `v4l_sys` reimported here
   |
   = note: `v4l_sys` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
74 | pub use v4l2_sys as other_v4l_sys;
   |                  ~~~~~~~~~~~~~~~~

error[E0425]: cannot find function `v4l2_open` in this scope
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/v4l2/api.rs:14:9
   |
14 |         v4l2_open(path, flags)
   |         ^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
9  +     use v4l_sys::v4l2_open;
   |

error[E0425]: cannot find function `v4l2_close` in this scope
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/v4l2/api.rs:17:9
   |
17 |         v4l2_close(fd)
   |         ^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
9  +     use v4l_sys::v4l2_close;
   |

error[E0425]: cannot find function `v4l2_ioctl` in this scope
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/v4l2/api.rs:27:9
   |
27 |         v4l2_ioctl(
   |         ^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
9  +     use v4l_sys::v4l2_ioctl;
   |

error[E0425]: cannot find function `v4l2_mmap` in this scope
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/v4l2/api.rs:44:9
   |
44 |         v4l2_mmap(
   |         ^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
9  +     use v4l_sys::v4l2_mmap;
   |

error[E0425]: cannot find function `v4l2_munmap` in this scope
  --> /home/barrett/.cargo/registry/src/index.crates.io-6f17d22bba15001f/v4l-0.14.0/src/v4l2/api.rs:54:9
   |
54 |         v4l2_munmap(start, length.try_into().expect("usize -> c size_t failed"))
   |         ^^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
9  +     use v4l_sys::v4l2_munmap;
   |

Some errors have detailed explanations: E0252, E0425, E0428.
For more information about an error, try `rustc --explain E0252`.
error: could not compile `v4l` (lib) due to 7 previous errors
warning: build failed, waiting for other jobs to finish...
barrett@farts ~/D/p/s/test_v4l_backend (main) [101]> 
```

**Comment by onkoe** [comment]
Also, please note that Cargo's features are intended to be additive. If you have two incompatible features, you may wish to add a `compile_error!` case when both are enabled: https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features

**Comment by MarijnS95** [comment]
This is somewhat documented at https://github.com/raymanfx/libv4l-rs?tab=readme-ov-file#dependencies. A `compile_error!()` would be nice to have (PR welcome!), but the "additive feature requirement" is a known `cargo` limitation for crate setups like this that _have_ to pick one or the other.

**Comment by onkoe** [comment]
Thanks - I'll look into a PR in the coming days!

For now, note that adding `default-features = false` alleviates the problem. That could be clear for experienced users, but just wanted to write it down for clarity. :)

**Comment by raymanfx** [comment]
Feel free to shoot a PR to implement the `compile_error!()` :)
